freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -269,24 +307,125 @@ def handle(self, event):
+        # compare replaced pullspecs with pullspecs in 'container_images' and
+        # create map for bundles that should be rebuilt with their nvrs
+        for container_image_nvr in self.event.container_images:
+            artifact_build = db.session.query(ArtifactBuild).filter(
+                ArtifactBuild.rebuilt_nvr == container_image_nvr,
+                ArtifactBuild.type == ArtifactType.IMAGE.value,
+            ).one_or_none()
+            pullspecs = []
+            # Try to find build in FM database, if it's not there check in Brew
+            if artifact_build:
+                pullspecs = artifact_build.bundle_pullspec_overrides["pullspecs"]
+            else:
+                # Fetch buildinfo from Koji
+                buildinfo = koji_api.get_build(container_image_nvr)
+                # Get the original pullspecs
+                pullspecs = (
+                    buildinfo.get('extra', {})
+                             .get('image', {})
+                             .get('operator_manifests', {})
+                             .get('related_images', {})
+                             .get('pullspecs', [])
+                )
+
+            for pullspec in pullspecs:
+                if pullspec.get('new') not in old_to_new_pullspec_map:
+                    continue
+                # use newer pullspecs in the image
+                pullspec['new'] = old_to_new_pullspec_map[pullspec['new']]
+                rebuild_nvr_to_pullspecs_map[container_image_nvr] = pullspecs
+
+        if not rebuild_nvr_to_pullspecs_map:
+            msg = 'None of the container images have applicable pullspecs ' \
+                  'from the input bundle images'
+            log.info(msg)
+            db_event.transition(EventState.SKIPPED, msg)
+            return []
+
+        # list with metadata about every bundle to do rebuild
+        to_rebuild_bundles = []
+        # fill 'append' and 'update' fields for bundles to rebuild
+        for nvr, pullspecs in rebuild_nvr_to_pullspecs_map.items():
+            bundle_digest = self._pyxis.get_manifest_list_digest_by_nvr(nvr)
+            if bundle_digest is not None:
+                bundles = self._pyxis.get_bundles_by_digest(bundle_digest)
+                temp_bundle = bundles[0]
+                csv_updates = (self._get_csv_updates(temp_bundle['csv_name'],
+                                                     temp_bundle['version']))
+                to_rebuild_bundles.append({
+                    'nvr': nvr,
+                    'update': csv_updates['update'],
+                    'pullspecs': pullspecs,
+                })
+            else:
+                log.warning('Can\'t find manifest_list_digest for bundle '
+                            f'"{nvr}" in Pyxis')
+
+        if not to_rebuild_bundles:
+            msg = 'Can\'t find digests for any of the bundles to rebuild'
+            log.warning(msg)
+            db_event.transition(EventState.SKIPPED, msg)
```

Comment:
As above, the state reason string need to be updated.

Reply:
@qixiang  But this situation is an error state. Shouldn't be there some other prefix? That the situation when none of the submitted `container_images` have digest in Pyxis, so it's not just 'No bundle to rebuild' situation.

Reply:
Is this an error? If yes, can we move it to `FAILED` instead. If freshmaker always move events to `FAILED` when there is error happens, you don't need to update these messages, release-driver can just ignore `state_reason` and only check the state.

Reply:
I am not sure. @mprahl what do you think? Is this an error, or just an unordinary situation?

Reply:
In my opinion, this is an error.

Reply:
Ok, so I will transition the event to the FAILED state, and remove prefix from the message